apps/cowswap-frontend/src/modules/tradeWidgetAddons/hooks/useRecipientValidation.ts

```diff
@@ -0,0 +1,124 @@
+import { useMemo } from 'react'
+
+import { isAddress } from '@cowprotocol/common-utils'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { useWalletInfo } from '@cowprotocol/wallet'
+
+import { RecipientRowProps } from 'modules/trade'
+
+import { resolveDisplayChainId } from 'common/utils/resolveDisplayChainId'
+
+export interface UseRecipientValidationParams {
+  recipient?: string | null
+  recipientEnsName?: string | null
+  recipientChainId?: number
+  account?: string | null
+  isFeeDetailsOpen?: boolean
```

Comment:
`isFeeDetailsOpen` doesn't have any relation to recipient validation, I don't think it should be mixed here